src/wc/wc.rs

```diff
@@ -56,18 +112,60 @@ impl Settings {
     }
 }
 
-struct Result {
-    title: String,
+#[derive(Debug, Default, Copy, Clone)]
+struct WordCount {
     bytes: usize,
     chars: usize,
     lines: usize,
     words: usize,
     max_line_length: usize,
 }
 
+impl Add for WordCount {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self {
+        Self {
+            bytes: self.bytes + other.bytes,
+            chars: self.chars + other.chars,
+            lines: self.lines + other.lines,
+            words: self.words + other.words,
+            max_line_length: max(self.max_line_length, other.max_line_length),
+        }
+    }
+}
+
+impl AddAssign for WordCount {
+    fn add_assign(&mut self, other: Self) {
+        *self = *self + other
+    }
+}
+
+impl WordCount {
+    fn with_title(self, title: String) -> TitledWordCount {
+        return TitledWordCount {
+            title: title,
+            count: self,
+        };
+    }
+}
+
+/// This struct supplements the actual word count with a title that is displayed
+/// to the user at the end of the program.
+/// The reason we don't simply include title in the `WordCount` struct is that
+/// it would result in unneccesary copying of `String`.
+#[derive(Debug, Default, Clone)]
+struct TitledWordCount {
```

Comment:
You can probably avoid the title allocation for the total at the end by changing this to:
```rust
#[derive(Debug, Default, Clone)]
struct TitledWordCount<‘a> {
    title: &’a str,
    count: WordCount,
}
```

I’ll try to take a more thorough look at the rest after work, but this looks good to me.

Reply:
Maybe we should let `title` be any type that implements `std::fmt::Display`, using some fancy generics? This code works as is though, we perhaps shouldn't complicate things further.

Reply:
It shouldn’t add much complexity to do that, so if you’d like to do so feel free.  I’m okay with it either way.